### PR TITLE
feat: add Profile search

### PR DESCRIPTION
**Pull request by rwmilford**
#### Summary of changes/features added:
- added profile searching
- moved onClick from fragment to adapters

#### Screenshots of UI changes (update with each new commit): 
image

## Pull Request Checklist
- [ ] Unit test written?
- [ ] (**For later in project**) Integration tests written?
- [ ] Java docs written?

### CODE REVIEW

File: client/app/src/main/java/com/example/criengine/Activities/RootActivity.java [hunk 1]
```diff
@@ -18,6 +18,11 @@
 import com.example.criengine.Fragments.NotificationFragment;
 import com.example.criengine.Fragments.RequestedBooksFragment;
 import com.example.criengine.Fragments.SearchBooksFragment;
+
+import com.example.criengine.Fragments.SearchBooksFragment;
+import com.example.criengine.Objects.Notification;
+import com.example.criengine.Objects.Profile;
+
```
Comment (ducklin5) [inline client/app/src/main/java/com/example/criengine/Activities/RootActivity.java:25]: we may not need these?

**Comment by AdityaHarvi** [comment]
Also, if all you're displaying for the user profile is the username, then you can make that list view smaller in comparison to the books listview cause the books is the one that has the main focus. So books should take up 2/3 of the screen while profile takes 1/3

File: client/app/src/main/java/com/example/criengine/Adapters/SearchProfilesListAdapter.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        final Profile profile = items.get(position);
+
+        // Sets current profile to xml field
+        searchUsername.setText(profile.getUsername());
+
+        // Opens to the book information screen when you click on a specific book.
+        view.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Intent intent = new Intent(v.getContext(), UserProfileActivity.class);
```
Comment (SamLojpur) [inline client/app/src/main/java/com/example/criengine/Adapters/SearchProfilesListAdapter.java:60]: All looks good, but user profile activity doesn't show books. Might need that to finish up but that may be outside the scope of this PR!

**Review by SamLojpur — APPROVED** [review]
Agree to make profiles list smaller, show only 2.5 profiles ish?

File: client/app/src/main/java/com/example/criengine/Activities/RootActivity.java (hunk 1 above)
Comment (rwmilford) [reply client/app/src/main/java/com/example/criengine/Activities/RootActivity.java:25]: removed them

File: client/app/src/main/java/com/example/criengine/Adapters/SearchProfilesListAdapter.java (hunk 2 above)
Comment (rwmilford) [reply client/app/src/main/java/com/example/criengine/Adapters/SearchProfilesListAdapter.java:60]: yeah, I'm gonna do it in another pr

**Pull request merged**